database/listingsNear.js

```diff
@@ -0,0 +1,17 @@
+const mongoose = require('mongoose');
+const db = require('./index.js');
+mongoose.Promise = global.Promise;
```

Comment:
depending on which version of Mongoose you're using, line 3 may not be necessary